org/postgresql/test/jdbc4/ArrayTest.java

```diff
@@ -311,4 +316,68 @@ public void testGetArrayOfComposites() throws SQLException {
 				fail("Needs to have 3 tokens");
 		}
     }
+
+    public void testCasingComposite() throws SQLException {
+        PGobject cc = new PGobject();
+        cc.setType("\"CorrectCasing\"");
+        cc.setValue("(1)");
+        Object[] in = new Object[1];
+        in[0] = cc;
+
+        Array arr = _conn.createArrayOf("\"CorrectCasing\"", in);
+        PreparedStatement pstmt = _conn.prepareStatement("SELECT ?::\"CorrectCasing\"[]");
+        pstmt.setArray(1, arr);
+        ResultSet rs = pstmt.executeQuery();
+
+        assertTrue(rs.next());
+        Object[] resArr = (Object[])rs.getArray(1).getArray();
+
+        assertTrue(resArr[0] instanceof PGobject);
+        PGobject resObj = (PGobject) resArr[0];
+        assertEquals("(1)", resObj.getValue());
+    }
+
+    public void testCasingBuiltinAlias() throws SQLException {
+        Array arr = _conn.createArrayOf("INT", new Integer[] { 1 , 2, 3});
```

Comment:
And this test checks the new behavior (which only worked with lowercase int before)